File: provider/src/main/java/com/raygun/raygun4android/network/RaygunOkHttpClientBuilder.java
package com.raygun.raygun4android.network;

import com.raygun.raygun4android.OkHttpClientBuilder;
import java.util.concurrent.TimeUnit;
import okhttp3.OkHttpClient;

public class RaygunOkHttpClientBuilder implements OkHttpClientBuilder {
    public static final int NETWORK_TIMEOUT = 30;

    public static RaygunOkHttpClientBuilder instance = new RaygunOkHttpClientBuilder();

    @Override
    public OkHttpClient build() {
        return new OkHttpClient.Builder()
                .connectTimeout(NETWORK_TIMEOUT, TimeUnit.SECONDS)
                .writeTimeout(NETWORK_TIMEOUT, TimeUnit.SECONDS)
                .readTimeout(NETWORK_TIMEOUT, TimeUnit.SECONDS)
                .build();
    }
}

File: provider/src/main/java/com/raygun/raygun4android/network/RaygunOkHttpClientBuilder.kt
package com.raygun.raygun4android.network

import com.raygun.raygun4android.OkHttpClientBuilder
import okhttp3.OkHttpClient
import java.util.concurrent.TimeUnit

class RaygunOkHttpClientBuilder : OkHttpClientBuilder {
    override fun build(): OkHttpClient =
        OkHttpClient
            .Builder()
            .connectTimeout(NETWORK_TIMEOUT.toLong(), TimeUnit.SECONDS)
            .writeTimeout(NETWORK_TIMEOUT.toLong(), TimeUnit.SECONDS)
            .readTimeout(NETWORK_TIMEOUT.toLong(), TimeUnit.SECONDS)
            .build()

    companion object {
        const val NETWORK_TIMEOUT: Int = 30

        @JvmField var instance: RaygunOkHttpClientBuilder = RaygunOkHttpClientBuilder()
    }
}
